### ISSUE TITLE
not possible to use system owl-lisp instead of downloading

### ISSUE THREAD

**Issue by anthraxx** [aoh/radamsa#25]
Hey,
I'm maintaining radamsa for a distribution and for building it would be nice to simply use the system owl-lisp and not download it from the internet.
In an earlier version i could just get rid of 'get-owl' target but not its quite strictly baked inside.

It would be gorgeous for distribution based packaging if we could have an optional switch or such to build without downloading owl-list and assume it being already installed

**Comment by aoh** [comment]
Radamsa started shipping with a prebuilt radamsa.c at 0.5, so release builds no longer depend on owl. 0.5 package is at https://github.com/aoh/radamsa/releases/download/v0.5/radamsa-0.5.tar.gz

If you want to build another version with an existing owl-lisp version, you can take a git checkout or a plain source package and build it with e.g. `$ make USR_BIN_OL=/usr/bin/ol build_radamsa`

Are these options suitable for packaging in your distribution?

**Comment by anthraxx** [comment]
Hmm not really, well the release does, but futute release will be different as the `Makefile` has changed and i want to be sure the next release will also work like that :smiley: :
it still downloads via wget: https://github.com/aoh/radamsa/blob/master/Makefile#L36
and uses `./bin/ol`: https://github.com/aoh/radamsa/blob/master/Makefile#L44

so what i need to do right now to make the git HEAD work:

```
  sed -r 's|(.*) bin/ol|\1|g' -i Makefile
  sed -r 's|bin/ol |/usr/bin/ol |g' -i Makefile
```

and then build via

```
  make \
    USR_BIN_OL=/usr/bin/ol \
    OWL="/usr/bin/ovm /var/lib/owl-lisp/fasl/init.fasl"
``
```

**Comment by aoh** [comment]
Argh. build_radamsa had gotten an accidental dependency on bin/ol.

Does `make USR_BIN_OL=/usr/bin/ol build_radamsa && make install` now work for you in HEAD? `build_radamsa` target exists to allow builds with an existing owl version.

**Comment by anthraxx** [comment]
@aoh 
well that works half way, during build it does not pull in via wget... but when i call install it does invoke that cascade :stuck_out_tongue_closed_eyes: 
see: https://github.com/aoh/radamsa/blob/master/Makefile#L47
intsall -> bin/radamsa -> radamsa.c -> bin/ol

PS: i think it would also make sense to everywhere pass ${LDFLAGS} where you use ${CFLAGS}, as its a standard for distributions to define custom flags there (like stack protector etc). Or doesn't that work woth ol?

**Comment by aoh** [comment]
Take two. radamsa.c now builds bin/ol if necessary, but doesn't depend on bin/ol. Better now? :)

Good thing you caught this.

**Comment by anthraxx** [comment]
fixed via fa95381ee42822a05df8f70654e9f54ade0dcbd9 thanks a lot :smile_cat: good we resolved this before the next release :yum:
